src/webapi/indexeddb.rs

```diff
@@ -0,0 +1,1145 @@
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/IDBCursor/continue)
+    fn next<K: Into<Option<Value>>>( &self, key: K) -> Result<(), IDBAdvanceError> {
+        match key.into() {
+            None => js_try!( @{self.as_ref()}.continue(); ).unwrap(),
+            Some(key) => js_try! ( @{self.as_ref()}.continue(@{key.as_ref()}); ).unwrap()
+        }
+    }
+
+    /// The continuePrimaryKey() method of the IDBCursor interface advances
+    /// the cursor to the to the item whose key matches the key parameter as
+    /// well as whose primary key matches the primary key parameter.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/IDBCursor/continuePrimaryKey)
+    fn continue_primary_key( &self, key: Value, primary_key: Value) -> Result<(), IDBContinuePrimaryKeyError> {
+        js_try!( @{self.as_ref()}.continuePrimaryKey(@{key}, @{primary_key}); ).unwrap()
+    }
+
+    /// The update() method of the IDBCursor interface returns an IDBRequest
+    /// object, and, in a separate thread, updates the value at the current
+    /// position of the cursor in the object store. If the cursor points to
+    /// a record that has just been deleted, a new record is created.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/IDBCursor/update)
+    fn update( &self, value: Value) -> Result<IDBRequest, IDBUpdateError> {
+        js_try!( return @{self.as_ref()}.update(@{value.as_ref()}); ).unwrap()
+    }
+
+    /// The delete() method of the IDBCursor interface returns an IDBRequest
+    /// object, and, in a separate thread, deletes the record at the cursor's
+    /// position, without changing the cursor's position. Once the record is
+    /// deleted, the cursor's value is set to null.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/IDBCursor/delete)
+    fn delete( &self ) -> Result<IDBRequest, IDBCursorDeleteError> {
+        js_try!( return @{self.as_ref()}.delete(); ).unwrap() 
+    }
+}
+
+/// The IDBCursor interface of the IndexedDB API represents a cursor for
+/// traversing or iterating over multiple records in a database.
+///
+/// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/IDBCursor)
+#[derive(Clone, Debug, PartialEq, Eq, ReferenceType)]
+#[reference(instance_of = "IDBCursor")]
+pub struct IDBCursor( Reference );
+
+impl IDBCursorSharedMethods for IDBCursor {}
+
+/// The IDBCursorWithValue interface of the IndexedDB API represents a cursor
+/// for traversing or iterating over multiple records in a database. It is
+/// the same as the IDBCursor, except that it includes the value property.
+///
+/// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/IDBCursorWithValue)
+#[derive(Clone, Debug, PartialEq, Eq, ReferenceType)]
+#[reference(instance_of = "IDBCursorWithValue")]
+pub struct IDBCursorWithValue( Reference );
```

Comment:
This is a subclass of `IDBCursor`. Please make it a subclass, similar to the case with `IDBOpenDBRequest`.